src/LndClient/LndTest.hs

```diff
@@ -311,31 +312,30 @@ syncPendingChannelsFor owner = this 0
             else mine1 (Proxy :: Proxy owner) >> this (attempt + 1)
 
 receiveClosedChannels ::
+  forall m owner.
   LndTest m owner =>
   Proxy owner ->
   [ChannelPoint] ->
-  TChan ((), ChannelEventUpdate) ->
   m (Either LndError ())
 receiveClosedChannels po = this 0
   where
-    this _ [] _ =
+    this _ [] =
       pure $ Right ()
-    this 30 _ _ =
+    this 30 _ =
       pure
         $ Left
         $ LndError "receiveClosedChannels - exceeded"
-    this (attempt :: Integer) cps0 cq = do
-      x0 <- readTChanTimeout (MicroSecondsDelay 1000000) cq
-      let x = channelEvent . snd <$> x0
-      $(logTM) InfoS $ logStr $
-        "receiveClosedChannels - got " <> (show x :: Text)
-      case x of
-        Just (ChannelEventUpdateChannelClosedChannel res) ->
-          case filter (/= chPoint res) cps0 of
-            [] -> pure $ Right ()
-            cps -> mine1 po >> this (attempt + 1) cps cq
-        _ ->
-          mine1 po >> this (attempt + 1) cps0 cq
+    this (attempt :: Integer) cps = do
+      let owners = enumerate :: [owner]
+      xsM <- mapM getOwnersCloseCPs owners
+      if all (\x -> x `elem` concat (rights xsM)) cps
```

Comment:
We can optimize this - not compute `concat (rights xsM)` every iteration. Also we need to be sure that 2 nodes have closed channel, not just 1